### PR TITLE
Migrate to new version of create-react-context or new React.createContext API

### PR DESCRIPTION
**Pull request by paul-phan**
I just have a bit confused why you need to wrap <StateContext.Consumer> in the Provider Component.

### CODE REVIEW

File: src/unstated.js [hunk 1]
```diff
@@ -119,23 +119,17 @@ export type ProviderProps = {
 };
 
 export function Provider(props: ProviderProps) {
-  return (
-    <StateContext.Consumer>
```
Comment (Andarist) [inline src/unstated.js]: you've removed really important feature of `unstated` here - nested `Provider`s, you must revert this

**Comment by ddombrow** [comment]
@Andarist Not the submitter, but I'm sitting here reading this trying to figure out why nested Providers are an important feature, and I haven't been successful.  Do you have an example where I might want to nest providers?

**Comment by jamiebuilds** [comment]
I'm not sure what you're trying to do with this PR since you did a whole bunch of separate things. Closing, reopen separate PRs.

**Comment by Andarist** [comment]
@ddombrow not `unstated` example but the idea is the same, look at [this](https://github.com/emotion-js/emotion/tree/master/packages/emotion-theming#themeprovider-reactcomponent)

With `unstated` u cannot adjust the nested provider with a function, but by using `inject` prop u can shadow some containers or add some.

**Pull request closed without merging**